include/circt/Dialect/ESI/ESIPorts.td

```diff
@@ -35,31 +35,35 @@ def ChannelType :
   Type<CPred<"$_self.isa<::circt::esi::ChannelPort>()">, "">;
 
 
-def Wrap : ESI_Op<"wrap", [NoSideEffect]> {
+//=========
+// Operations on ports.
+
+def WrapValidReady : ESI_Op<"wrap.vr", [NoSideEffect]> {
```

Comment:
Just a questing about the naming with the dot here: is the idea to expose other ways to wrap/unwrap ports as other `wrap.xxx` operations down the line? This case is wrapping the port with valid and ready, so it is `wrap.vr`, but later, if there is some other fancy signaling wrapping the port, it would be `wrap.fancy`?

Reply:
Yep. There are all sorts of control signaling methods which will have to be represented as an attribute on the ChannelPort type and in the wrap/unwrap ops. For instance, in high-frequency designs, modules may want to expose their internal buffers and use them as run-out buffers for feed-forward, credit-controlled communications. Those would be esi.wrap.fifo or some such name.